File: backend/src/models/BannerImage.php
<?php
/**
 * BannerImage Model
 */

namespace MilalHomepage\Models;

use MilalHomepage\Utils\Database;

class BannerImage {
    private $db;

    public function __construct() {
        $this->db = Database::getInstance()->getConnection();
    }

    public function getActive() {
        try {
            $stmt = $this->db->prepare("
                SELECT * FROM banner_image
                WHERE is_active = 1
                ORDER BY id DESC
                LIMIT 1
            ");
            $stmt->execute();
            return $stmt->fetch() ?: null;
        } catch (\PDOException $e) {
            error_log("Get banner image error: " . $e->getMessage());
            return null;
        }
    }
}

File: backend/app/routers/banner_image.py
from fastapi import APIRouter, Depends
from pymysql.connections import Connection

from ..database import get_db, serialize
from ..response import success

router = APIRouter(prefix="/banner-image", tags=["banner-image"])


@router.get("")
def get_active(db: Connection = Depends(get_db)):
    with db.cursor() as cur:
        cur.execute(
            "SELECT * FROM banner_image WHERE is_active = 1 ORDER BY id DESC LIMIT 1"
        )
        row = cur.fetchone()
    return success(serialize(row))
